plugin/cancel/cancel.py

```diff
@@ -1,20 +1,70 @@
 # to disable command cancellation, comment out this entire file.
 # you may also wish to adjust the commands in misc/cancel.talon.
-from talon import actions, speech_system
+
+import time
+
+from talon import Context, Module, actions, speech_system
+from talon.grammar import Phrase
 
 # To change the phrase used to cancel commands, you must also adjust misc/cancel.talon
 cancel_phrase = "cancel cancel".split()
 
+mod = Module()
+ctx = Context()
+
+ts_threshold: float = 0
+
+
+@ctx.action_class("speech")
+class SpeechActions:
+    # When Talon wakes we set the timestamp threshold. On the next command we
+    # will compare the phrase timestamp to the threshold and cancel any phrase
+    # started before wakeup. This is to prevent speech said before wake-up to
+    # be interpreted as a command if the user wakes Talon using a noise or
+    # keypress.
+    def enable():
+        actions.user.cancel_current_phrase()
+        actions.next()
+
+
+@mod.action_class
+class Actions:
+    def cancel_current_phrase():
+        """Cancel/abort current spoken phrase"""
+        global ts_threshold
+        ts_threshold = time.perf_counter()
 
-def pre_phrase(d):
-    n = len(cancel_phrase)
-    if "text" in d and "parsed" in d:
-        before, after = d["text"][:-n], d["text"][-n:]
-        if after != cancel_phrase:
+
+def pre_phrase(phrase: Phrase):
+    global ts_threshold
+
+    # Nothing to cancel
+    if "parsed" not in phrase:
+        return
+
+    # Check if the phrase is before the threshold
+    if ts_threshold != 0:
+        words = phrase["phrase"]
+        start = getattr(words[0], "start", phrase["_ts"])
+        phrase_starts_before_threshold = start < ts_threshold
+        ts_threshold = 0
+        # Start of phrase is before threshold timestamp
+        if phrase_starts_before_threshold:
+            abort_phrase(phrase)
```

Comment:
Consider printing this phrase to the log? It's possible some people would *want* the other behavior; it may be helpful to see what is being discarded so they can disable this if needed. That's what my version does (I didn't realize you were going to submit this PR so rapidly or I wouldn't have spent the time!): https://github.com/nriley/talon_community/blob/480449064a77399948b7491b2b790619c14db262/core/modes/abort.py#L36

Reply:
Print added